src/components/Button/button.module.scss

```diff
@@ -155,15 +155,39 @@
         }
     }
 
+    .counter {
+        background-color: var(--button-counter-default-background-color);
+        display: inline-block;
+        padding: 0 $space-xxs;
+        text-align: center;
+        width: fit-content;
+    }
+
     &:disabled,
     &.disabled {
         opacity: $disabled-alpha-value;
         cursor: not-allowed;
     }
 
+    &:hover {
+        .counter {
+            background-color: var(--button-counter-hover-background-color);
+        }
+    }
+
+    &:active {
+        .counter {
+            background-color: var(--button-counter-active-background-color);
+        }
+    }
+
```

Comment:
instead of overriding badge styles, can we use the badge active state ?
and disruptive for diruptive button